jetty-annotations/src/main/java/org/eclipse/jetty/annotations/AnnotationParser.java

```diff
@@ -561,31 +564,26 @@ public AnnotationParser(int javaPlatform, int asmVersion)
      */
     public void addParsedClass (String classname, Resource location)
     {
-        List<String> list = new ArrayList<>(1);
-        if (location != null)
-            list.add(location.toString());
-
-        List<String> existing = _parsedClassNames.putIfAbsent(classname, list);
+        String existing = _parsedClassNames.putIfAbsent(classname, location.toString());
```

Comment:
I would save the call to `toString()` here, and use a `Map<String, Resource>`.

Reply:
But the point is to output the log warning, which wants the resource as a string. So I've solved it a different way by doing location.toString() once.